modules/plugins/youtube.go

```diff
@@ -488,3 +664,567 @@ func (yt *YouTube) newMsg(content string) *discordgo.MessageSend {
+
+	_, err = query.Run(helpers.GetDB())
+	return
+}
+
+func (yt *YouTube) deleteEntry(id string) (n int, err error) {
+	query := rethink.Table(youtubeDbTableName).Filter(rethink.Row.Field("id").Eq(id)).Delete()
+
+	r, err := query.RunWrite(helpers.GetDB())
+	if err == nil {
+		n = r.Deleted
+	}
+
+	return
+}
+
+// yt.lock must held.
+func (yt *YouTube) runYoutubeFeedsLoop() {
+	if yt.feedsLoopRunning {
+		yt.logger().Error("youtube feeds loop already running")
+		return
+	}
+	yt.feedsLoopRunning = true
+
+	go yt.youtubeFeedsLoop()
+
+	return
+}
+
+func (yt *YouTube) youtubeFeedsLoop() {
+	defer helpers.Recover()
+	defer func() {
+		yt.Lock()
+		yt.feedsLoopRunning = false
+		yt.Unlock()
+
+		go func() {
+			yt.logger().Error("The checkYoutubeFeedsLoop died. Please investigate! Will be restarted in 60 seconds")
+			time.Sleep(60 * time.Second)
+
+			yt.Lock()
+			yt.runYoutubeFeedsLoop()
+			yt.Unlock()
+		}()
+	}()
+
+	if err := yt.quota.Init(yt); err != nil {
+		helpers.Relax(err)
+	}
+
+	for ; ; time.Sleep(10 * time.Second) {
+		err := yt.quota.UpdateCheckingInterval()
+		helpers.Relax(err)
+
+		yt.checkYoutubeFeeds()
+	}
+}
+
+func (yt *YouTube) checkYoutubeFeeds() {
+	yt.RLock()
+	defer yt.RUnlock()
+
+	t := time.Now().Unix()
+	entries, err := yt.readEntries(rethink.Row.Field("next_check_time").Le(t))
+	helpers.Relax(err)
+
+	for _, e := range entries {
+		switch e.ContentType {
+		case "channel":
+			e = yt.checkYoutubeChannelFeeds(e)
+		case "quota":
+			continue
+		default:
+			yt.logger().Error("unknown contents type: " + e.ContentType)
+		}
+
+		// update next check time
+		e = yt.setNextCheckTime(e)
+		err := yt.updateEntry(e)
+		helpers.Relax(err)
+	}
+}
+
+func (yt *YouTube) checkYoutubeChannelFeeds(e DB_Youtube_Entry) DB_Youtube_Entry {
+	c, err := e.getChannelContent()
+	if err != nil {
+		yt.logger().WithFields(logrus.Fields{
+			"id": e.ID,
+		}).Error(err)
+		return e
+	}
+
+	// set iso8601 time which will be used search query filter "published after"
+	lastSuccessfulCheckTime := time.Unix(e.LastSuccessfulCheckTime, 0)
+	publishedAfter := lastSuccessfulCheckTime.
+		Add(time.Duration(-1) * time.Hour).
+		Format("2006-01-02T15:04:05-0700")
+
+	// get updated feeds
+	feeds, err := yt.getChannelFeeds(c.ID, publishedAfter)
+	if err != nil {
+		yt.logger().Error("check channel feeds error: " + err.Error() + " channel name: " + c.Name + "id: " + e.ID)
+		return e
+	}
+
+	newPostedVideos := make([]string, 0)
+	alreadyPostedVideos := make([]string, 0)
+
+	// check if posted videos and post new videos
+	for i := len(feeds) - 1; i >= 0; i-- {
+		feed := feeds[i]
+
+		// only 'upload' type video can be posted
+		if feed.Snippet.Type != "upload" {
+			continue
+		}
+		videoId := feed.ContentDetails.Upload.VideoId
+
+		// check if the video is already posted
+		if yt.isPosted(videoId, c.PostedVideos) {
+			alreadyPostedVideos = append(alreadyPostedVideos, videoId)
+			continue
+		}
+
+		// make a message and send to discord channel
```

Comment:
Can we make our own embed using video name, video thumbnail and channel name? 
Something like the other SNS services have, example from V Live:
![screen shot 2017-10-07 at 21 50 37](https://user-images.githubusercontent.com/1156226/31311300-98df80dc-aba9-11e7-907d-ad2971932c44.png)
https://github.com/Seklfreak/Robyul2/blob/06bbf5a85680f5882592d97e294e2e2afd6be2c5/modules/plugins/vlive.go#L583-L591 
https://github.com/Seklfreak/Robyul2/blob/06bbf5a85680f5882592d97e294e2e2afd6be2c5/modules/plugins/vlive.go#L596-L599
Content:
`<youtube video link>`
Embed: 
Title:`🎥  Channel Name uploaded a new video!`
URL: Link to video
Description: `**Video Title**`
Thumbnail: Channel Icon
Image: Video Thumbnail
Description `powered by youtube.com`
Something like that. This will make it more easily to understand what Robyul just posted.

Reply:
How about this?

![screenshot](https://user-images.githubusercontent.com/6683987/31314071-7b989152-ac31-11e7-8b3d-f07f47df9df1.png)

Sadly, youtube Activity API calls doesn't give a channel image URL. If channel icon is needed, then we can store in database by using one more query when channel is added. However I'm not sure how to handle when URL is changed.

Reply:
That's perfect! I don't think it's worth to go through channel image URL hassle, having the video thumbnail is the important part. 👍